### ISSUE TITLE
[Python] Implement new Concept Exercise: str methods

### ISSUE THREAD

**Issue by DavidGerva** [exercism/v3#2175]
This issue describes how to implement the `str` method concept exercise for the Python track, which should explain how basic `str` methods work for text processing in python.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The python sequence type `str` contains a number of [useful methods](https://docs.python.org/3.7/library/stdtypes.html#string-methods) for text processing and this concept exercise should make the student aware of those methods, and teach how to use them with strings to return new (changed) strings.

## Learning objectives

- become familiar with the `str` class, and its core methods
- understand when it is appropriate to use  methods of `str`  to operate on a string
- use several of the methods from each grouping below to operate on strings:
    - check if a given string ends/begins with the a specified suffix/prefix with the `endswith()`/`startswith()` methods
    - check if a given string is composed of certain code points/characters by using `isalnum()`, `isalph()`, `isascii()`, `isdecimal()`,  `isdigit()`, or `isnumeric()` methods.
   -  return a new string that is a transform of the input string using `capitalize()`, `lower()`, `casefold()`, `swapcase()`, `upper()`, or `title()` methods
  - return a new string that has leading/trailing whitespaces removed with `lstrip()`, `rstrip()`, or `strip()`methods
  - check if a given substring is contained within a given string using `in` 
  - return a new string that replaces a given substring with a new value using the `replace()` method
  - build a new string from an input `iterable` with `join()`

> **_NOTE:_**  the above is  just a subset of methods: it is not necessary that you use every single one in the creation of the exercise. Please also feel free to add any other `str` methods you deem important for student understanding.

## Out of scope

-  string formatting methods,  "f-strings", and "format strings" (_they will have their own exercise_)
-  string splitting 
-  string translate methods (_will have their own exercise_)
- `String` constants from the `string` module
- `built-ins` that can take a `str` as an argument
-  encoding/decoding & codecs
-  the binary sequence types bytes, bytearray, and memoryview (_they will have their own exercise_)
- binary transforms and text transforms
- text encodings

## Concepts

- `str`
- methods
- immutable/immutability
- unicode code point

## Prerequisites

- `booleans`
- `strings`
- `sequences`, `sequence type`
- `iteration`

## Resources to refer to

- [Python str type (Python language reference)](https://docs.python.org/3.7/library/stdtypes.html#text-sequence-type-str)
- [Python str methods](https://docs.python.org/3.7/library/stdtypes.html#string-methods)
- [Python string methods on journaldev](https://www.journaldev.com/24588/python-string-functions)

## Hints

- Referring to one or more of the resources linked above, or analogous resources from a trusted source.

## Concept Description 
(_a variant of this can be used for the `v3/languages/python/concepts/<concept>.md` doc and `introduction.md`_)

A `str` in Python is an immutable **_sequence_** of Unicode code points which can include letters, numbers, symbols, punctuation, etc.   Strings implement all of the [common sequence operations](https://docs.python.org/3/library/stdtypes.html#typesseq-common), along with iteration using the `for item in <string>` syntax.

Python provides many convenient [string methods](https://docs.python.org/3/library/stdtypes.html#string-methods) for cleaning, splitting, translating, and otherwise working with the `str` type.   New strings can be created based on method arguments, and/or additional information can be returned. Strings can be concatenated using the `+` operator or by using `str.join()`.

Because strings are immutable (_a string's value doesn't change_); methods that appear to modify a string actually return a new instance of `str`.

## Representer

No changes required.

## Analyzer

No changes required.

## Implementing

Tests should be written using unittest.TestCase and the test file named comparisons_test.py.

Code in the `.meta/example.py` file should only use syntax & concepts introduced in this exercise or one of its prerequisites. 
Please do not use comprehensions, generator expressions, or other syntax not previously covered. Please also follow [PEP8](https://www.python.org/dev/peps/pep-0008/) guidelines.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by kimolivia** [comment]
Hi, I would like to work on this issue. Is this still okay to work on? (I noticed help-wanted was removed)

**Comment by BethanyG** [comment]
Hi @kimolivia - we're in the process of re-working these issues and concepts, so we're "on hold" for a bit, which is why the "help wanted" labels were removed.  But we would ❤️   some contribution help!  We should be up and ready for contributors by Halloween/first week of November, so please check back then.

**Comment by BethanyG** [comment]
@kimolivia  -- if you are still up for writing up a concept exercise, this one is ready for you!  Just give a shout if you are still interested, or if you have any questions or issues.

**Comment by OMEGA-Y** [comment]
Hi, I would like to work on this issue. It is labeled by good first issue.
But, I couldn't find which section I can help although reading contents.
Please let me know or guide.

**Comment by kimolivia** [comment]
@BethanyG yes, I would love to take this issue! I'll reach out with any questions!

**Comment by BethanyG** [comment]
Hi @OMEGA-Y,

Thanks so much for stepping forward and volunteering!  ❤️  🎉  

This issue has already been claimed by @kimolivia (_she expressed interest before we re-wrote the issue_) - but we're going to be putting up more today -- and in the days to come.  _We've got a lot of issues of this type we would LOVE help on!_

Most of the issues in this repo describe new _**concept exercises**_ (_programing problems_) for different programming languages that need to be created/written for the next version of the [exercism website](https://exercism.io/).  If you are familiar with the current version of exercism, these practice problems are a variation of the ones there -- except they will be focused on a _single concept in the programming language_ like `loops`, or `str-methods`, or `classes`.  

Each exercise we're making includes an introduction to the concept, a story/problem, example code for a solution, a related test suite that a student can use to test their solution to the problem, and a set of links the student can use to learn more.  There are other files -- but those are the main ones we need to create.  Each issue is intended to give an exercise writer the pieces they need to start writing those files.

I want to apologize ahead of time for our current documentation - you've caught us in a bit of a re-org with how our concept exercises are put together, so the issues and the current structure don't exactly match up with the documentation.  We are working on updating everything this week.  All that being said, the easiest way to get started is with the existing documentation links and video at the top of each issue:

1.  [concept exercises](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md)
2.  [what are the new features?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md) 
3.  [why are we changing things?](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md)

[video on concept exercises](https://www.youtube.com/watch?v=gkbBqd7hPrA)

We're also very happy to chat and help with ideas and setup in our Slack channel.  My username is @BethanyG  there, and I would be very happy to walk you through how to fork the repo and get started.  😄

**Comment by OMEGA-Y** [comment]
@BethanyG Thanks for answering! I don't know if I can ask more here...
Is it possible to add a new exercise to the problem-specification repository now? Or is it better to make a new concept exercise in this repo now?
And can you tell me the Slack channel url?

**Comment by BethanyG** [comment]
Hi @OMEGA-Y  - apologies for the delay.  I've sent an invite to Slack to the email listed on your GitHub profile.  Let me know if there is a different email you'd like me to use.  It might be better to ask additional questions there or in a new issue, so that we don't spam kimolivia with our discussions.  😄 

The problem-specification repository is for exercises we've made/are making for the _current version_ of the [exercism website](https://exercism.io/), and for the more open-ended "practice" problems that will be in place once we switch to V3.  If you have an idea/improvement to those exercises, you would make it in that repository.  We'd be delighted if you wanted to propose a new exercise for the Python track (or another track) there!

**This** repository is for the new concept exercises for V3, and for now it is a combination of all language tracks.  I know we're a little behind getting issues up for Python in particular -- if you have a subject you really want to take on, we can always discuss it in Slack and then I can put up an issue for you.  But there are also issues available in other languages/tracks like Type Script, JavaScript, C, Ruby, etc.  We may not have labeled many "good first issues" -- but if you ask questions on those you are interested in, I am sure that one of the maintainers would be happy to answer any questions you have.

**Comment by BethanyG** [comment]
@kimolivia - do you need an invite to our Slack channel?  Just let me know if you do need one.  😄

**Comment by BethanyG** [comment]
Closing as this exercise has been merged successfully.  Thank you @kimolivia and @valentin-p !  🎉
